src/locallookup.h

```diff
@@ -36,30 +36,27 @@ class LocalLookup : public Lookup
      */
     const Hosts &_hosts;
     
-    /**
-     *  Is the operation ready?
-     *  @var bool
-     */
-    bool _ready = false;
-
     /**
      *  Method that is called when it is time to process this lookup
      *  @param  now     current time
-     *  @return bool    should it be rescheduled?
+     *  @return bool    was there a call to userspace?
      */
     virtual bool execute(double now) override
     {
         // do nothing if ready
-        if (_ready) return false;
-        
-        // pass to the hosts
-        _hosts.notify(Request(this), _handler, this);
+        if (_handler == nullptr) return false;
```

Comment:
```suggestion
        if (finished()) return false;
```

In fact, I think this should be asserted: `assert(!finished())` as it seems like a programming error to not check this before calling execute.

Reply:
I find this difficult. It is true that from the perspective of the entire library this method should indeed not be called when the lookup is already finished. But from the perspective of just this class (or just this method) we prepare for the worst and also handle cases that there is no crash even when it _is_ called...